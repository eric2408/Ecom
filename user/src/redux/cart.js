import {createSlice} from '@reduxjs/toolkit';


const cart = createSlice({
    name:"cart",
    initialState:{
        products: [],
        quantity:0,
        total:0
    },
    reducers:{
        addProduct: (state, action) => {
            state.quantity += action.payload.quantity;
            state.products.push(action.payload)
            state.total += action.payload.price * action.payload.quantity;
        },
        deleteProduct: (state, action) => {
            state.products= [];
            state.quantity= 0;
            state.total= 0;
        },
        addOne: (state, action) => {
            state.quantity += 1;
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0) {
                state.products[index].quantity += 1;
            }
        },
        subtractOne: (state, action) => {
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0 && state.products[index].quantity > 0) {
                state.products[index].quantity -= 1;
                state.quantity -= 1;
            }
        },
        removeOne:(state, action) => {
            const index = state.products.findIndex(
                    (item) => item._id === action.payload.id
                );
            if (index >= 0) {
                    state.quantity -= state.products[index].quantity;

                    state.products.splice(index, 1);
            } else {
                    console.warn(
                      `Cant remove product (id: ${action.payload.id}) as its not in basket!`
                    )
            }
        }
    }
})

export const {addProduct,  deleteProduct, addOne, subtractOne, removeOne} = cart.actions

export default cart.reducer;

export const emptyCart = async(dispatch) => {
    try{
      dispatch(deleteProduct());
    } catch(e){
      console.log(e)
    }
  }

