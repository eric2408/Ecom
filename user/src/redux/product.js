import { createSlice } from "@reduxjs/toolkit";
import { userRequest } from "../requestAxios";

export const productSlice = createSlice({
  name: "product",
  initialState: {
    products: [],
    isFetching: false,
    error: false,
  },
  reducers: {
    //Get all products
    getProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    getProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products = action.payload;
    },
    getProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Delete Products
    deleteProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    deleteProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.splice(state.products.findIndex((item) => item._id === action.payload), 1);
    },
    deleteProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Update products
    updateProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    updateProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products[
        state.products.findIndex((item) => item._id === action.payload.id)
      ] = action.payload.product;
    },
    updateProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Add Product
    addProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    addProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.push(action.payload);
    },
    addProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
  },
});

export const {
  getProductStart,
  getProductSuccess,
  getProductFailure,
  deleteProductStart,
  deleteProductSuccess,
  deleteProductFailure,
  updateProductStart,
  updateProductSuccess,
  updateProductFailure,
  addProductStart,
  addProductSuccess,
  addProductFailure,
} = productSlice.actions;

export default productSlice.reducer;

export const getProducts = async (dispatch) => {
    dispatch(getProductStart());
    try {
      const res = await userRequest.get("/products");
      dispatch(getProductSuccess(res.data));
    } catch (err) {
      dispatch(getProductFailure());
    }
  };


export const deleteProduct = async (id, dispatch) => {
  dispatch(deleteProductStart());
  try {
    const res = await userRequest.delete(`/products/${id}`);
    dispatch(deleteProductSuccess(id));
  } catch (err) {
    dispatch(deleteProductFailure());
  }
};

export const updateProduct = async (id, product, dispatch) => {
  dispatch(updateProductStart());
  try {
    const res = await userRequest.put(`/products/${id}`, product);
    dispatch(updateProductSuccess({ id, product }));
  } catch (err) {
    dispatch(updateProductFailure());
  }
};
export const addProduct = async (product, dispatch) => {
  dispatch(addProductStart());
  try {
    const res = await userRequest.post(`/products`, product);
    dispatch(addProductSuccess(res.data));
  } catch (err) {
    dispatch(addProductFailure());
  }
};