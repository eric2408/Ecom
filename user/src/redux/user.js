import { createSlice } from "@reduxjs/toolkit";
import { generalRequest } from "../requestAxios";

const user = createSlice({
  name: "user",
  initialState: {
    currentUser: null,
    isFetching: false,
    error: false,
  },
  reducers: {
    loginStart: (state) => {
      state.isFetching = true;
    },
    loginSuccess: (state, action) => {
      state.isFetching = false;
      state.error = false;
      state.currentUser = action.payload;
    },
    loginFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
    logOut: (state) => {
      state.currentUser = null;
    },
  },
});

export const { loginStart, loginSuccess, loginFailure, logOut } = user.actions;
export default user.reducer;


export const login = async(dispatch, user) =>{
    dispatch(loginStart());
    try{
        const res = await generalRequest.post('/auth/login', user)
        dispatch(loginSuccess(res.data))
    } catch(e){
        dispatch(loginFailure())
    }
}

export const logout = async(dispatch, user) => {
  try{
    dispatch(logOut());
  } catch(e){
    console.log(e)
  }
}

