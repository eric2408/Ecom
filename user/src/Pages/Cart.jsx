import React from 'react';
import styled from "styled-components";
import Announcement from '../Components/Announcement';
import Footer from '../Components/Footer';
import Navbar from '../Components/Navbar';
import RemoveIcon from '@mui/icons-material/Remove';
import AddIcon from '@mui/icons-material/Add';
import { Image } from '@mui/icons-material';
import { mobileScreen } from '../Helper';
import cart from '../redux/cart';
import { useSelector } from "react-redux";
import StripeCheckout from "react-stripe-checkout";
import { useEffect, useState } from "react";
import { userRequest } from "../requestAxios";
import { useNavigate } from 'react-router-dom';
import robo from '../img/sq.png';
import { addOne, subtractOne, removeOne } from '../redux/cart';
import { useDispatch } from 'react-redux';

const publicKey = "pk_test_51LkzBrHoffqkiVupSdU4Fbe8SwocQzAnycibSRBvY7llyFe8jqxflhDqhxikcwItfswSa5lk8a0ewH2vld9BzsmB00U4HdhCNB";

const Container = styled.div`
`;

const Padding = styled.div`
  padding: 50px;
  margin-bottom: 100px;
  ${mobileScreen({ padding: "10px" })}
`;

const Title = styled.h1`
  font-weight: 300;
  text-align: center;
  margin-bottom: 100px;
`;


const Orders = styled.div`
  display: flex;
  justify-content: space-between;
  ${mobileScreen({ flexDirection: "column" })}
`;

const Information = styled.div`
  flex: 3;
`;

const Product = styled.div`
margin-bottom: 50px;
  display: flex;
  justify-content: space-between;
  ${mobileScreen({ flexDirection: "column" })}
`;

const Detail = styled.div`
  flex: 2;
  display: flex;
`;

const Img = styled.img`
  width: 200px;
`;

const Description = styled.div`
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
`;

const Name = styled.span``;

const Id = styled.span``;

const Color = styled.div`
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background-color: ${(props) => props.color};
`;

const Size = styled.span``;

const Price = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const AmountContainer = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 20px;
`;

const Amount = styled.div`
  font-size: 24px;
  margin: 5px;
  ${mobileScreen({ margin: "5px 15px" })}
`;

const ProductPrice = styled.div`
  font-size: 30px;
  font-weight: 200;
  ${mobileScreen({ marginBottom: "20px" })}
`;

const Hr = styled.hr`
  background-color: #eee;
  border: none;
  height: 1px;
`;

const OrderSummary = styled.div`
  flex: 1;
  border: 0.5px solid lightgray;
  border-radius: 10px;
  padding: 20px;
  height: 50vh;
`;

const STitle = styled.h1`
  font-weight: 200;
`;

const SItem = styled.div`
  margin: 30px 0px;
  display: flex;
  justify-content: space-between;
  font-weight: ${(props) => props.type === "total" && "500"};
  font-size: ${(props) => props.type === "total" && "24px"};
`;

const SItemText = styled.span``;

const SItemPrice = styled.span``;



const Button = styled.button`
  width: 100%;
  padding: 10px;
  background-color: black;
  color: white;
  font-weight: 600;
  cursor: pointer;
`;

const ButtonTwo = styled.button`
  margin-top: 20px;
  padding: 15px;
  border: 2px solid #6f50e6;
  background-color: white;
  cursor: pointer;
  font-weight: 500;
  &:hover {
    background-color: #6f50e6;
    color: white;
  }
`

const Cart = () => {
  const cart = useSelector(state => state.cart);
  console.log(cart)
  const [token, setToken] = useState(null);
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const onToken = (token) => {
    setToken(token)
  }

  useEffect(()=> {
    const req = async () => {
     try{
      const response = await userRequest.post('/checkout/payment', {
        tokenId: token.id,
        amount: cart.total *100
      });
      navigate('/success', { state: {stripeData: response.data, products: cart }});
     } catch (e) {
      console.log(e)
     } 
    };

    token && req();
  }, [token, cart.total, navigate])


  const handleClick = (sign, id) => {
      if(sign === 'addition'){
            dispatch(addOne({id}))
    
      } else if(sign === 'subtract'){
            dispatch(subtractOne({id}))
      }
  }

  const removeFromBasket = (id) => {
    dispatch(removeOne({ id }))
  }

  return (
    <Container>
      <Announcement />
      <Navbar />
      <Padding>
        <Title>Shopping Cart</Title>
        <Orders>
          <Information>
            {cart.products.map(product => (
            <Product>
              <Detail>
                <Img src={product.img} />
                <Description>
                  <Name>
                    <b>Product:</b> {product.title}
                  </Name>
                  <Id>
                    <b>ID:</b> {product._id}
                  </Id>
                  <Color color={product.color} />
                  <Size>
                    <b>Size:</b> {product.size}
                  </Size>
                </Description>
              </Detail>
              <Price>
                <AmountContainer>
                  <AddIcon onClick={()=> handleClick('addition', product._id)}/>
                  <Amount>{product.quantity}</Amount>
                  <RemoveIcon onClick={()=> handleClick('subtract', product._id)}/>
                </AmountContainer>
                <ProductPrice>$ {cart.total}</ProductPrice>
                <ButtonTwo onClick={()=> removeFromBasket(product._id)}>Remove</ButtonTwo>
              </Price>

            </Product>))}
            <Hr />
          </Information>
          <OrderSummary>
            <STitle>ORDER SUMMARY</STitle>
            <SItem>
              <SItemText>Subtotal</SItemText>
              <SItemPrice>$ {cart.total}</SItemPrice>
            </SItem>
            <SItem>
              <SItemText>Estimated Shipping</SItemText>
              <SItemPrice>$ 3.50</SItemPrice>
            </SItem>
            <SItem>
              <SItemText>Shipping Discount</SItemText>
              <SItemPrice>$ -3.50</SItemPrice>
            </SItem>
            <SItem type="total">
              <SItemText>Total</SItemText>
              <SItemPrice>$ {cart.total}</SItemPrice>
            </SItem>
            <StripeCheckout
              name="Virtual Space"
              image={robo}
              billingAddress
              shippingAddress
              description={`Your total is $${cart.total}`}
              amount={cart.total * 100}
              token={onToken}
              stripeKey={publicKey}
            >
              <Button>CHECKOUT NOW</Button>
            </StripeCheckout>
          </OrderSummary>
        </Orders>
      </Padding>
      <Footer />
    </Container>
  )
}

export default Cart