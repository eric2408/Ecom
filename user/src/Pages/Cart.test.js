import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import {store, persistor} from '../redux/store';
import{Provider} from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import Cart from './Cart'

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <BrowserRouter>
        <Cart />
      </BrowserRouter>
    </PersistGate>
  </Provider>

  );
  expect(asFragment()).toMatchSnapshot();
});