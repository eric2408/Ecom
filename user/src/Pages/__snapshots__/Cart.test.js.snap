// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-cwSeag gempnc"
  >
    <div
      class="sc-bcXHqe eLQayE"
    >
      Free Shipping on Orders Over $49!
    </div>
    <div
      class="sc-eDWCr lhYMNP"
    >
      <div
        class="sc-bqWxrE gvnzob"
      >
        <div
          class="sc-ksBlkl cxUMGU"
        >
          <span
            class="sc-hBxehG ddBtzW"
          >
            EN
          </span>
          <div
            class="sc-fnGiBr dKybLJ"
          >
            <input
              class="sc-fEXmlR jPVLHP"
              placeholder="search"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              style="color: white; margin-left: 8px; font-size: 20px; cursor: pointer;"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-bjfHbI kLHHpZ"
        >
          <a
            href="/"
            style="text-decoration: none;"
          >
            <h1
              class="sc-idXgbr csUYOV"
            >
              VIRTUAL SPACE
            </h1>
          </a>
        </div>
        <div
          class="sc-dIfARi ckAuwl"
        >
          <a
            href="/register"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-dmctIk hHlIX"
            >
              <div
                class="sc-hHTYSt giXObl"
              >
                REGISTER
              </div>
               
            </div>
          </a>
          <a
            href="/login"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hHTYSt giXObl"
            >
              SIGN IN
            </div>
          </a>
          <a
            href="/cart"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hHTYSt giXObl"
            >
              <span
                class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ShoppingCartIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary BaseBadge-badge BaseBadge-invisible css-fu0map-MuiBadge-badge"
                />
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-lllmON htGZHw"
    >
      <h1
        class="sc-iJnaPW jYoojy"
      >
        Shopping Cart
      </h1>
      <div
        class="sc-gikAfH hgCEuS"
      >
        <div
          class="sc-ezOQGI eZhDOI"
        >
          <hr
            class="sc-eeMvmM jNempU"
          />
        </div>
        <div
          class="sc-cUEOzv djpcdL"
        >
          <h1
            class="sc-jOiSOi elWgKC"
          >
            ORDER SUMMARY
          </h1>
          <div
            class="sc-dPWrhe dKybuT"
          >
            <span
              class="sc-bCfvAP"
            >
              Subtotal
            </span>
            <span
              class="sc-cOxWqc"
            >
              $ 0
            </span>
          </div>
          <div
            class="sc-dPWrhe dKybuT"
          >
            <span
              class="sc-bCfvAP"
            >
              Estimated Shipping
            </span>
            <span
              class="sc-cOxWqc"
            >
              $ 3.50
            </span>
          </div>
          <div
            class="sc-dPWrhe dKybuT"
          >
            <span
              class="sc-bCfvAP"
            >
              Shipping Discount
            </span>
            <span
              class="sc-cOxWqc"
            >
              $ -3.50
            </span>
          </div>
          <div
            class="sc-dPWrhe iefjzY"
            type="total"
          >
            <span
              class="sc-bCfvAP"
            >
              Total
            </span>
            <span
              class="sc-cOxWqc"
            >
              $ 0
            </span>
          </div>
          <button
            class="sc-AHaJN iRhaBs"
          >
            CHECKOUT NOW
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-gswNZR jofBsk"
    >
      <div
        class="sc-dkrFOg kLzEQW"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
          <h1
            class="sc-hLBbgP cfrXqI"
          >
            VIRTUAL SPACE
          </h1>
        </a>
        <p
          class="sc-eDvSVe jvdQks"
        >
          Join Virtual Space! Time go a fascinating world!
        </p>
        <div
          class="sc-jSUZER bHEyKf"
        >
          <div
            class="sc-gKPRtg bThFsx"
            color="4267B2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FacebookIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
              />
            </svg>
          </div>
          <div
            class="sc-gKPRtg eqicWh"
            color="cd486b"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="InstagramIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
              />
            </svg>
          </div>
          <div
            class="sc-gKPRtg fQhJD"
            color="1DA1F2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TwitterIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-iBYQkv jBFHGg"
      >
        <h3
          class="sc-ftTHYK gFQvOI"
        >
          Support
        </h3>
        <ul
          class="sc-pyfCe bBXOyA"
        >
          <li
            class="sc-jrcTuL fBSxcv"
          >
            About Virtual Space
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Account
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Orders
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Favorites
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Cart
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            Privacy Policy
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            Refund Policy
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            Terms of Service
          </li>
        </ul>
      </div>
      <div
        class="sc-kDvujY hcROd"
      >
        <h3
          class="sc-ftTHYK gFQvOI"
        >
          Contact Us
        </h3>
        <div
          class="sc-ipEyDJ hXqFPG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LocationOnIcon"
            focusable="false"
            style="margin-right: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           27 San Carlos St. Buffalo, NY 14215
        </div>
        <div
          class="sc-ipEyDJ hXqFPG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PhoneIphoneIcon"
            focusable="false"
            style="margin-right: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 1h-8C6.12 1 5 2.12 5 3.5v17C5 21.88 6.12 23 7.5 23h8c1.38 0 2.5-1.12 2.5-2.5v-17C18 2.12 16.88 1 15.5 1zm-4 21c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm4.5-4H7V4h9v14z"
            />
          </svg>
           +1 234 56 7899
        </div>
        <div
          class="sc-ipEyDJ hXqFPG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            style="margin-right: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
           jo123@gmail.com
        </div>
        <img
          class="sc-csuSiG dabsRR"
          src="https://chargie.org/wp-content/uploads/2021/03/PayPal-stripe-web-opt-300x136-1-300x136-300x136-1.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
